### ISSUE TITLE
istio_tcp_sent_bytes_total and istio_tcp_received_bytes_total are swapped

### ISSUE THREAD

**Issue by ldemailly** [istio/istio#32399]
**Bug description**
TCP bytes sent on outgoing connection and received are reversed

This is a screen shot of sending request at 50 qps and receiving 2k responses: (mislabelling a fortio http echo as tcp on purpose to demonstrate the issue:)

![image](https://user-images.githubusercontent.com/3664595/115777410-18087380-a36a-11eb-9691-b6dbb57b2ce0.png)

It's not just Grafana but the actual prometheus metrics istio_tcp_sent_bytes_total and istio_tcp_received_bytes_total are swapped

[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[x] Extensions and Telemetry
[ ] Security
[x] Test and Release
[x] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

Bytes sent to be sent and received to be received (from the perspective of the pod; seems to be maybe from perspective of envoy)

**Steps to reproduce the bug**

See description

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

1.8.5

**How was Istio installed?**

helm

**Environment where the bug was observed (cloud vendor, OS, etc)**

AWS EKS

**Comment by bianpengyuan** [comment]
Indeed looks like a bug.. Bytes received / sent logic should be direction aware: https://github.com/istio/proxy/blob/e02cbbe2e3f390afd9711c5d5e9f4bb5462059ea/extensions/stats/plugin.h#L369-L378

**Comment by ldemailly** [comment]
cc @douglas-reid

**Comment by kyessenov** [comment]
This is working as intended. @ldemailly the confusion is from the fact that we report from the perspective of the proxy. The metrics should be interpreted bytes sent from _source_ to _destination_ as reported by _reporter_. In that sense, the metrics are intentionally almost identical from both sides.

**Comment by ldemailly** [comment]
cough cough. hi kuat, no you can't quite say that bytes sent on outgoing connection is actually bytes received :-)

**Comment by kyessenov** [comment]
hi Laurent. I think the chart needs to be better explained. The outgoing connection is not what most assume (it's not from pod, it's to the proxy).

**Comment by ldemailly** [comment]
all the http sent/received are consistent except for tcp, it's not an explanation it's inconsistent/ a bug

**Comment by ldemailly** [comment]
In other words, I understand that's true from
the proxy's point of view but that doesn't mean it's the right name. Istio is about then workloads, the bytes the workload receives and sends; same as for http. the proxy is "transparent"

**Comment by rolandkool** [comment]
Hi

I also ran into this issue when building a custom dashboard. Using the [hipstershop demo app](https://github.com/GoogleCloudPlatform/microservices-demo) I noticed that the traffic between the cartservice and the redis service that it uses is the same on both sides. So both sender and receiver report the same about of bytes for sent and received, which does not make any sense. Sure, it can be solved in a grafana dashboard by swapping the metrics, but I think it makes more sense to fix this in the proxy itself.
